### PR TITLE
Replace check using maybe check

### PR DESCRIPTION
**Pull request by luqiang-guo**

### CODE REVIEW

File: oneflow/user/ops/partial_fc_sample_op.cpp [hunk 1]
```diff
@@ -63,10 +63,11 @@ REGISTER_USER_OP("distributed_partial_fc_sample")
       return Maybe<void>::Ok();
     })
     .SetInputArgModifyFn([](user_op::GetInputArgModifier GetInputArgModifierFn,
-                            const user_op::UserOpConfWrapper&) {
+                            const user_op::UserOpConfWrapper&) -> Maybe<void> {
       user_op::InputArgModifier* label_modifier = GetInputArgModifierFn("label", 0);
       CHECK_NOTNULL(label_modifier);
```
Comment (daquexian) [inline oneflow/user/ops/partial_fc_sample_op.cpp:68]: 这些 CHECK_NOTNULL 也需要改成 CHECK_NOTNULL_OR_RETURN

File: oneflow/user/ops/reshape_like_op.cpp [hunk 2]
```diff
@@ -40,10 +40,11 @@ REGISTER_USER_OP("reshape_like")
       return Maybe<void>::Ok();
     })
     .SetInputArgModifyFn([](user_op::GetInputArgModifier GetInputArgModifierFn,
-                            const user_op::UserOpConfWrapper&) {
+                            const user_op::UserOpConfWrapper&) -> Maybe<void> {
       user_op::InputArgModifier* like_modifier = GetInputArgModifierFn("like", 0);
       CHECK_NOTNULL(like_modifier);
```
Comment (liufengwei0103) [inline oneflow/user/ops/reshape_like_op.cpp:45]: CHCEK_NOTNULL_OR_RETURN

File: oneflow/user/ops/pad2d_ops.cpp [hunk 3]
```diff
@@ -269,10 +271,11 @@ REGISTER_USER_OP("constant_pad2d")
     })
     .SetGetSbpFn(GetOpSbpSignature)
     .SetInputArgModifyFn([](user_op::GetInputArgModifier GetInputArgModifierFn,
-                            const user_op::UserOpConfWrapper&) {
+                            const user_op::UserOpConfWrapper&) -> Maybe<void> {
       user_op::InputArgModifier* x_modifier = GetInputArgModifierFn("x", 0);
       CHECK_NOTNULL(x_modifier);
```
Comment (liufengwei0103) [inline oneflow/user/ops/pad2d_ops.cpp:276]: CHCEK_NOTNULL_OR_RETURN

File: oneflow/user/ops/pad2d_ops.cpp [hunk 4]
```diff
@@ -175,10 +176,11 @@ REGISTER_USER_OP("replication_pad2d")
...
```
Comment (liufengwei0103) [inline oneflow/user/ops/pad2d_ops.cpp:181]: CHCEK_NOTNULL_OR_RETURN

File: oneflow/user/ops/pad2d_ops.cpp [hunk 5]
```diff
@@ -83,10 +83,11 @@ REGISTER_USER_OP("reflection_pad2d")
...
```
Comment (liufengwei0103) [inline oneflow/user/ops/pad2d_ops.cpp:88]: 好像CHECK_NOTNULL都漏了，再注意一下这个吧

**Review by liufengwei0103 — COMMENTED** [review]
还有漏改的op:
./cast_to_tick_op.cpp:40:          CHECK_EQ(in_dis_hint.sbp_parallel_size(), parallel_hierarchy.NumAxes());
./conv_op.cpp:26:  CHECK_EQ(NDims + 2, in->shape().NumAxes());
 ./conv_op.cpp:76:    CHECK_EQ(weight->shape(), Shape(weight_shape));
 ./conv_op.cpp:167:  CHECK_EQ(ndims, strides.size());
 ./conv_op.cpp:168:  CHECK_EQ(ndims, dilation_rate.size());
  ./deconv_op.cpp:26:  CHECK_EQ(NDims + 2, in->shape().NumAxes());
再检查一下

**Review by liufengwei0103 — COMMENTED** [review]
https://github.com/Oneflow-Inc/oneflow/blob/500386b2f7087636a9ccd5e9a9def190dff97400/oneflow/core/operator/user_op.cpp#L515 比如这里，需要加JUST()

File: oneflow/user/kernels/assign_if_kernel.cpp [hunk 6]
```diff
@@ -27,17 +27,24 @@ class AssignIfCPUKernel final : public user_op::OpKernel {
   ~AssignIfCPUKernel() override = default;
 
  private:
-  void Compute(user_op::KernelComputeContext* ctx) const override {
+  Maybe<void> Compute(user_op::KernelComputeContext* ctx) const override {
     const user_op::Tensor* condition = ctx->Tensor4ArgNameAndIndex("condition", 0);
-    if ((assign_if == (*condition->dptr<C>() == 0))) { return; }
+    if ((assign_if == (*condition->dptr<C>() == 0))) {
+      return Maybe<void>::Ok();
+      ;
```
Comment (liufengwei0103) [inline oneflow/user/kernels/assign_if_kernel.cpp:34]: 这里貌似多一个分号

File: oneflow/user/kernels/assign_if_kernel.cu [hunk 7]
```diff
@@ -33,19 +33,23 @@ class AssignIfGPUKernel final : public user_op::OpKernel {
...
+  Maybe<void> Compute(user_op::KernelComputeContext* ctx) const override {
     const user_op::Tensor* condition = ctx->Tensor4ArgNameAndIndex("condition", 0);
     CHECK_EQ(condition->shape().NumAxes(), 1);
     CHECK_EQ(condition->shape().At(0), 1);
     const user_op::Tensor* value = ctx->Tensor4ArgNameAndIndex("value", 0);
     user_op::Tensor* ref = ctx->Tensor4ArgNameAndIndex("ref", 0);
-    if (value->dptr() == ref->dptr()) { return; }
+    if (value->dptr() == ref->dptr()) {
+      return Maybe<void>::Ok();
+      ;
```
Comment (liufengwei0103) [inline oneflow/user/kernels/assign_if_kernel.cu:44]: 同上

File: oneflow/user/kernels/assign_kernel.cpp [hunk 8]
```diff
@@ -26,10 +26,13 @@ class AssignKernel final : public user_op::OpKernel {
...
 
  private:
-  void Compute(user_op::KernelComputeContext* ctx) const override {
+  Maybe<void> Compute(user_op::KernelComputeContext* ctx) const override {
     const user_op::Tensor* value_tensor = ctx->Tensor4ArgNameAndIndex("value", 0);
     user_op::Tensor* ref_tensor = ctx->Tensor4ArgNameAndIndex("ref", 0);
-    if (value_tensor->dptr() == ref_tensor->dptr()) { return; }
+    if (value_tensor->dptr() == ref_tensor->dptr()) {
+      return Maybe<void>::Ok();
+      ;
```
Comment (liufengwei0103) [inline oneflow/user/kernels/assign_kernel.cpp:34]: 多分号第一次编译，会警报出来。可以关注一下。

**Review by liufengwei0103 — APPROVED** [review]

**Pull request closed without merging**